### PR TITLE
Fix ilasm parsing of defines

### LINKED ISSUE #42001: Unable to consume latest dotnet/runtime
### Description

Windows Forms 6.0 build is failing unable to take latest updates from dotnet/runtime
	
* PR: https://github.com/dotnet/winforms/pull/3804
* Build: https://dev.azure.com/dnceng/public/_build/results?buildId=805275&view=logs&jobId=1f132584-ab08-5fbf-e5d0-5685de454342&j=1f132584-ab08-5fbf-e5d0-5685de454342&t=1ce98a45-1f3f-5914-0114-7502bfbbe8a4
	 

```
##[error]local_define(1,0): error : (NETCORE_ENGINEERING_TELEMETRY=Build) syntax error at token '┐' in: ╗∩System.Runtime 

error MSB3073: (NETCORE_ENGINEERING_TELEMETRY=Build) The command "F:\workspace\_work\1\s\.packages\\runtime.win-x64.microsoft.netcore.ilasm\6.0.0-alpha.1.20454.24\runtimes\win-x64\native/ilasm -QUIET -NOLOGO -RESOURCES=F:\workspace\_work\1\s\artifacts\obj\Accessibility\Debug\netcoreapp5.0\Accessibility.ref.res.obj -DLL -OUTPUT="F:\workspace\_work\1\s\artifacts\obj\Accessibility\Debug\netcoreapp5.0\Accessibility.dll" -KEY="F:\workspace\_work\1\s\.packages\microsoft.dotnet.arcade.sdk\5.0.0-beta.20452.19\tools\snk/35MSSharedLib1024.snk" Accessibility.il" exited with code 1.
```

### Regression?

Possibly?

### PR DESCRIPTION
**Pull request by jkotas**
Regression introduced by #41611

Fixes #42001

### CODE REVIEW

**Comment by jkotas** [comment]
cc @am11

**Comment by am11** [comment]
Ah, so in this a case of ANSI string, which we discussed not to support / care about in ilasm, but it turned out we need to support it.

**Comment by am11** [comment]
Ah, so in this a case of ANSI string, which we decided not to support / care about in ilasm, but it turned out we do need to support it.

File: src/coreclr/src/ilasm/grammar_after.cpp [hunk 1]
```diff
@@ -924,10 +924,6 @@ int yylex()
                                     tok = parse_literal(curSym, curPos, FALSE);
                                     if(tok == QSTRING)
                                     {
-                                        // insert UTF-8 BOM prefix
```
Comment (janvorli) [inline src/coreclr/src/ilasm/grammar_after.cpp:927]: Will strings encoded in UTF-8 still work?

Comment (jkotas) [reply src/coreclr/src/ilasm/grammar_after.cpp:927]: I was not able to figure out what this was supposed to help with. Non-ASCII QSTRINGs (quoted strings) never worked in this position as far as I can tell. The following:

```
#define NAME "ššš"

.assembly NAME
```

fails in all .NET Framework and earlier .NET Core versions of ilasm. The case that this change is fixing is:

```
#define NAME "sss"

.assembly NAME
```

Comment (janvorli) [reply src/coreclr/src/ilasm/grammar_after.cpp:927]: Ok, thanks for the details.

**Review by janvorli — APPROVED** [review]

**Comment by JulieLeeMSFT** [comment]
Are we going to backport it to RC2? or Will this be .NET6 fix?

**Comment by jkotas** [comment]
This is .NET 6 only regression. No backporting necessary.

**Pull request merged**